src/lib.rs

```diff
@@ -55,14 +54,23 @@ pub struct Action {
     pub operation: FileOperation,
 }
 
+#[derive(Debug, Clone, Default)]
+pub struct Ctx {
+    pub cmd_opt: ApplyOpts,
+}
+
 pub fn process(ctx: &Ctx) -> Result<()> {
     debug!("extracting variables from context",);
-    let (confirmed_variables, suggested_variables) = ctx::extract_variables(ctx)?;
+    let (mut default_variables, mut confirmed_variables, suggested_variables) =
```

Comment:
I prefer that they are not `mut`, and to create a new Variables than combine them. More clearer to understand and debug.